tests/warn/t10373.scala

```diff
@@ -1,4 +1,4 @@
-//> using options -Xfatal-warnings -deprecation -feature
+//> using options -deprecation -feature
 
 abstract class Foo {
   def bar(): Unit = this match {
```

Comment:
Why doesn't `bar` also get a non-exhaustive warning as the `baz` below?

Reply:
I agree we shouldn't check exhaustivity for `Foo`. In this sense, the selector type is not sealed, then matching any subtype of it would be considered as "(maybe) covering the whole space". 

Therefore, the tuple arguments should also cover the whole space. `Tuple2` is sealed only means the tuple itself need to be covered, and the content should not be affected, so we should not check exhaustivity for the tuple arguments here as well.

Reply:
Yes, because the two matching do look identical to me.

Reply:
I hesitate to do that, as I fear it would be classified a "regression" that we're no longer warning there.  I understand the position the dilemma, though.